py7zr/compressor.py

```
@@ -1173,6 +1175,13 @@ def needs_password(cls, coders) -> bool:
                 return True
         return False
 
+    @classmethod
+    def needs_password_for_filters(cls, filters) -> bool:
+        for filter in filters:
```

Ruff (0.15.17)

[error] Variable `filter` is shadowing a Python builtin

(A001)